From: Michael Froman <mfroman@mozilla.com>
Date: Mon, 20 Jun 2022 22:34:00 +0000
Subject: Bug 1775143 - pt2 - apply reconciling patch to match previous import
 r=ng

tl;dr We commit extra changes when we vendored this for the v86 import,
so we need to do the equivalent here again.  For more info see:
https://wiki.mozilla.org/Media/WebRTC/libwebrtc_Update_Process/automation_plan#verify_current_vendoring_scripts_cmd-line_params
and
https://wiki.mozilla.org/File:Vendor-build-fixup.patch.zip

(skip-generation)

Depends on D149807

Differential Revision: https://phabricator.services.mozilla.com/D149816
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/6125938ac1986d298aeb169e364d75cc8368a441
---
 config/BUILD.gn          |  2 +-
 toolchain/linux/BUILD.gn | 23 +++++++++++++++++++++++
 2 files changed, 24 insertions(+), 1 deletion(-)

diff --git a/config/BUILD.gn b/config/BUILD.gn
index ed94a16a0..9eed1913f 100644
--- a/config/BUILD.gn
+++ b/config/BUILD.gn
@@ -133,7 +133,7 @@ config("debug") {
     # Enable libstdc++ debugging facilities to help catch problems early, see
     # http://crbug.com/65151 .
     # TODO(phajdan.jr): Should we enable this for all of POSIX?
-    defines += [ "_GLIBCXX_DEBUG=1" ]
+    #defines += [ "_GLIBCXX_DEBUG=1" ]
   }
 }
 
diff --git a/toolchain/linux/BUILD.gn b/toolchain/linux/BUILD.gn
index 6e11d5e8c..ce8004ebd 100644
--- a/toolchain/linux/BUILD.gn
+++ b/toolchain/linux/BUILD.gn
@@ -246,6 +246,13 @@ clang_toolchain("clang_mips64el") {
   }
 }
 
+clang_toolchain("clang_mips64") {
+  toolchain_args = {
+    current_cpu = "mips64"
+    current_os = "linux"
+  }
+}
+
 gcc_toolchain("mipsel") {
   toolprefix = "mipsel-linux-gnu-"
 
@@ -335,6 +342,22 @@ gcc_toolchain("s390x") {
   }
 }
 
+gcc_toolchain("mips64") {
+  cc = "gcc"
+  cxx = "g++"
+
+  readelf = "readelf"
+  nm = "nm"
+  ar = "ar"
+  ld = cxx
+
+  toolchain_args = {
+    current_cpu = "mips64"
+    current_os = "linux"
+    is_clang = false
+  }
+}
+
 gcc_toolchain("ppc64") {
   cc = "gcc"
   cxx = "g++"
